pkg/module/kubebuilder/project.go

```diff
@@ -144,30 +145,53 @@ func (p *Project) DefaultCR(s step.Step) ([]byte, error) {
 	}
 
 	samplesDir := filepath.Join(p.path, samplesPath)
-	d, err := os.ReadDir(samplesDir)
+	filez, err := os.ReadDir(samplesDir)
```

Comment:
Another thing we can improve, at the moment, this readDir will basically read all files in the directory. But if it contains file other than yaml, it will also list them as options. This actually not a edge case, for example, if user edits CR files with vim, but quit unexpected, then it will have a temporary .swp file generated, like the example below:
```
2023-02-23T10:26:35.969+0100	DEBUG	module/resources.go:179	Kubebuilder project detected.
? Please specify the file to use as default CR in  template-operator/config/samples:
[1] .operator.kyma-project.io_v1alpha1_samplehelm.yaml.swp
[2] operator.kyma-project.io_v1alpha1_samplehelm.yaml
```

So I suggest implement some logic to filter out those non `yaml` or `yml` file.

Reply:
Is that filter really necessary? It seems that the function has already bin expanded a lot. Extra logic might affect the readability of the code.

Reply:
This has been now implemented. User Experience >>